controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java

```diff
@@ -453,31 +453,32 @@ private boolean servicesConverged(ApplicationId id, JobType type, DualLogger log
     private void sendNotification(Run run, DualLogger logger) {
         Application application = controller.applications().require(run.id().application());
         Notifications notifications = application.deploymentSpec().notifications();
-        if (notifications != Notifications.none()) {
-            boolean newCommit = application.change().application()
-                                           .map(run.versions().targetApplication()::equals)
-                                           .orElse(false);
-            When when = newCommit ? failingCommit : failing;
-
-            List<String> recipients = new ArrayList<>(notifications.emailAddressesFor(when));
-            if (notifications.emailRolesFor(when).contains(author))
-                run.versions().targetApplication().authorEmail().ifPresent(recipients::add);
-
-            try {
-                if (run.status() == outOfCapacity && run.id().type().isProduction())
-                    controller.mailer().send(mails.outOfCapacity(run.id(), recipients));
-                if (run.status() == deploymentFailed)
-                    controller.mailer().send(mails.deploymentFailure(run.id(), recipients));
-                if (run.status() == installationFailed)
-                    controller.mailer().send(mails.installationFailure(run.id(), recipients));
-                if (run.status() == testFailure)
-                    controller.mailer().send(mails.testFailure(run.id(), recipients));
-                if (run.status() == error)
-                    controller.mailer().send(mails.systemError(run.id(), recipients));
-            }
-            catch (RuntimeException e) {
-                logger.log(INFO, "Exception trying to send mail for " + run.id(), e);
-            }
+        boolean newCommit = application.change().application()
+                                       .map(run.versions().targetApplication()::equals)
+                                       .orElse(false);
+        When when = newCommit ? failingCommit : failing;
```

Comment:
This seems to be something closer to an `Event`? Consider renaming.

That would also make for nicer signatures imo, e.g. `Notifications#emailAddressesFor(Event)`.

Reply:
It represents an XML attribute called `when`, and I think the two are similar enough it's OK to keep this name.